contentful/apps
# feat: Improve ColorPicker Field

## Summary
   - Replace deprecated `FieldExtensionSDK`;
   - Replace deprecated `AppExtensionSDK`;
   - Display only validated colors in a validated field;
   - Show hex value in dropdown.

## Purpose


The purpose is to improve UX for the content editor by simplifying the color picker field.

## Approach


I have also considered exposing a boolean in the configuration screen to validate against the label (`white` instead of `#ffffff`) but that would introduce breaking changes. I am thinking about supporting both and converting label to value behind the scenes, but I will leave this for a future PR.

## Testing steps


1. Create two text fields with the Color Picker appearance
2. In one of the fields, add validation for some of your theme values
3. Confirm that in the validated field's dropdown you only see the validated items available for selection

## Breaking Changes


App should still function as previously; if there is no validation present, `validatedColors` return the `theme.colors` array unchanged.

## Dependencies and/or References


N/A

## Deployment


N/A

## Screenshots
### Unvalidated Color Picker field
![image](https://github.com/contentful/apps/assets/69549795/c86f7a38-0423-4e21-91de-4a72aa104444)

### Validated Color Picker field
![image](https://github.com/contentful/apps/assets/69549795/4327ab68-5d84-4305-b51a-0779be7e8d68)
